explainer-beforematch.md

```diff
@@ -3,21 +3,25 @@
 ### Summary (TL;DR)
 
 The `beforematch` event allows developers to display content to the user in
-response to find-in-page or fragment navigation actions. It is fired just
-before these actions, and at render timing. The applicable actions are the
-following:
-
-* There is a new find-in-page active match, in which case the event is fired
-  with the matched element as the target.
-* There is a fragment-link navigation, in which case the event is fired with
-  the matched fragment as the target.
-* There is a scroll-to-text navigation, in which case the event is fired with
+response to actions which scroll the page to a target element. It is fired
+at render timing just before these actions scroll the page. The applicable
+actions are the following:
+
+* There is a new find-in-page (ctrl+f) active match, in which case the event
+  is fired with the matched element as the target.
+* There is an element fragment navigation (`example.com/#foo`) or the fragment
+  is changed by script (`window.location.hash = ...`), in which case the event
+  is fired with the matched element as the target.
+* There is a [scroll-to-text](https://github.com/WICG/ScrollToTextFragment)
+  navigation (`example.com/#:~:text=foo`), in which case the event is fired with
   the matched element as the target.
 
 Note that the 'matched element' in this document refers to one of the following:
-* the element that contains the text which was matched by find-in-page or
-  scroll-to-text navigation.
-* the element which was selected for the fragment link navigation.
+* In the find-in-page and scroll-to-text cases: The closest block-level element
```

Comment:
```suggestion
* In the find-in-page and scroll-to-text cases: The closest block-level ancestor element
```

Also, I would specify here that if the matched text spans multiple block-level elements, then the event is fired on the block-level ancestor of the beginning of the match
(possibly worded better)

Reply:
matching text spanning multiple block level elements is only possible with scroll-to-text-fragment right? in that case, didn't we earlier discuss making it only fire on the first of the two block level elements? I guess we could change it to fire on the ancestor...

Reply:
You're right, I meant to say that there should be two changes here:
- Specify that it's the closest block-level **ancestor** element (_ie clarify what "closest" means by saying that we're talking about ancestor distance here_)
- Specify that if it spans multiple blocks, then use **the beginning of the match** as the spot from which you look for the ancestor element

Sorry if that wasn't clear. I am _not_ suggesting a **common** ancestor for the range of the match. Does that make sense?

Reply:
got it, thanks!

Reply:
How does the latest patch look?